trimesh/visual/texture.py

```diff
@@ -19,10 +19,18 @@ def uv_to_color(uv, image):
     colors : (n, 4) float
       RGBA color at each of the UV coordinates
     """
-    x = (uv[:, 0] * image.width).round().astype(int)
-    y = ((1 - uv[:, 1]) * image.height).round().astype(int)
+    # https://shapenet.org/qaforum/index.php?qa=15&qa_1=why-is-the-texture-coordinate-in-the-obj-file-not-in-the-range  # NOQA
```

Comment:
it seems opengl supports uv values out of range from [0, 1] by truncating that int value, for example:
- 1.1 -> 0.1
- -0.3 -> 0.7

Reply:
Ah that's a little weird, we may want to just clamp them to the range ourselves for the sake of explicitness, like `np.clip(uv, a_min=0.0, a_max=1.0)`

Reply:
clip makes

- 1.1 -> 1
- -0.3 -> 0

right? This is not what we expect.

Reply:
I guess it depends on what you expect haha, that behavior seems a bizzare to me but if it's what people do then we should probably replicate it, through something like:
```
In [12]: uv[uv < 0.0] += 1.0
In [13]: uv %= 1.0
```

Reply:
Actually, uv can be under -1 like -21.3 for shapenet models.
Meshlab still can load that model, so it is still supposed to be an acceptable format.

Reply:
Ah good point. Looks like normalizing is not a good idea: https://stackoverflow.com/questions/49923442/threejs-normalize-uv-coordinates-to-0-and-1-issue

Reply:
Actually don't we just want to mod-1 the values right before evaluating? I.E: 
```
x = ((uv[:, 0] % 1.0) * (image.width - 1)).round().astype(np.int64)
y = (((1 - uv[:, 1]) % 1.0) * (image.height - 1)).round().astype(np.int64)
```